airbyte-ci/connectors/base_images/README.md

```diff
@@ -6,21 +6,22 @@ Our connector build pipeline ([`airbyte-ci`](https://github.com/airbytehq/airbyt
 Our base images are declared in code, using the [Dagger Python SDK](https://dagger-io.readthedocs.io/en/sdk-python-v0.6.4/).
 
 - [Python base image code declaration](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connectors/base_images/base_images/python/bases.py)
-- ~Java base image code declaration~ _TODO_
+- ~Java base image code declaration~ *TODO* 
 
-## Where are the Dockerfiles?
 
+## Where are the Dockerfiles?
 Our base images are not declared using Dockerfiles.
 They are declared in code using the [Dagger Python SDK](https://dagger-io.readthedocs.io/en/sdk-python-v0.6.4/).
 We prefer this approach because it allows us to interact with base images container as code: we can use python to declare the base images and use the full power of the language to build and test them.
 However, we do artificially generate Dockerfiles for debugging and documentation purposes.
 
-### Example for `airbyte/python-connector-base`:
 
+
+### Example for `airbyte/python-connector-base`:
 ```dockerfile
-FROM docker.io/python:3.9.18-slim-bookworm@sha256:44b7f161ed03f85e96d423b9916cdc8cb0509fb970fd643bdbc9896d49e1cad0
+FROM docker.io/python:3.9.19-slim-bookworm@sha256:b92e6f45b58d9cafacc38563e946f8d249d850db862cbbd8befcf7f49eef8209
 RUN ln -snf /usr/share/zoneinfo/Etc/UTC /etc/localtime
-RUN pip install --upgrade pip==23.2.1
+RUN pip install --upgrade pip==24.0 setuptools==70.0.0
```

Comment:
I don't like that we have the versions hardcoded here in code. But, not blocking, definitely works for us well now.

Reply:
We hardcode versions for build reproducibility. If we don't do that a rebuild without any change might end up with different versions... Which can be surprising and lead to unexpected side effets.